### ISSUE TITLE
Type name missing for typed test

### ISSUE THREAD

**Issue by ads00** [onqtam/doctest#120]
Hello,

Type name is not displayed in the report for templated tests.
It would be cool to have `type_to_string` displaying some information when type is not manually registerd.
Or maybe replace `doctest::Types` by a recursive macro to register types.
```
// create doctest::Types + register types
DOCTEST_TYPES(Ts, my_type1, my_type2);

SCENARIO_TEMPLATE("test", T, Ts)

// or directly
SCENARIO_TEMPLATE("test", T, my_type1, my_type2)
```

**Comment by onqtam** [comment]
Good idea - perhaps a recursive macro would be the best option. I'm still knee-deep in other work so this will have to wait... but I'd be happy to accept an initial implementation - will speed things up!

**Comment by mitchgrout** [comment]
Based off the original description it sounds like this behaviour is effectively implemented.
